### PR TITLE
Add default verb support

### PR DESCRIPTION
**Pull request by Artentus**
I added basic support for a default verb (a verb that will be assumed if no verb is specified).
Tests all completed on my end but I'm not too knowledgable about this code, if there is any problem let me know.

### CODE REVIEW

**Comment by moh-hassan** [comment]
Thanks @Artentus for your work.
I did a quick test:
- for a default verb, it works as expected.
- With more than default verb (not allowed), an Exception "Sequence contains more than one matching element"  is fired.
 You can validate that only one verb has a default and fire error,e.g,  "More than one default verb is not allowed"
- Modify the HelpText screen "--help" to show the help with indication of the default verb, e.g
 >read       (Default verb) Reads from source.
 write      Writes to destination.

 - Add test cases to cover the above cases.

**Comment by Artentus** [comment]
The error with more than one default verb is expected. However I do agree a more descriptive error message could be useful.

As for the help text, I'll first have to see how that is actually implemented before I can modify it. Will see what I can do.

**Comment by moh-hassan** [comment]
>As for the help text, I'll first have to see how that is actually implemented before I can modify it. Will see what I can do.

You can concentrate in the error handling and its unit test and push the modification, so I can start review the PR.

**Comment by Artentus** [comment]
Done, there is now proper error handling and a couple of new unit tests.

File: src/CommandLine/VerbAttribute.cs [hunk 1]
```diff
@@ -12,19 +12,22 @@ namespace CommandLine
...
         /// </summary>
         /// <param name="name">The long name of the verb command.</param>
-        /// <exception cref="System.ArgumentException">Thrown if <paramref name="name"/> is null, empty or whitespace.</exception>
-        public VerbAttribute(string name)
+        /// <param name="isDefault">Whether the verb is the default verb.</param>
+        /// <exception cref="System.ArgumentException">Thrown if <paramref name="name"/> is null, empty or whitespace and <paramref name="isDefault"/> is false.</exception>
+        public VerbAttribute(string name, bool isDefault = false)
         {
-            if (string.IsNullOrWhiteSpace(name)) throw new ArgumentException("name");
+            if (!isDefault && string.IsNullOrWhiteSpace(name)) throw new ArgumentException("name");
```
Comment (moh-hassan) [inline src/CommandLine/VerbAttribute.cs]:
```suggestion
            if (string.IsNullOrWhiteSpace(name)) throw new ArgumentException("name");
```

File: src/CommandLine/Core/Verb.cs [hunk 2]
```diff
@@ -12,12 +12,17 @@ sealed class Verb
         private readonly string name;
         private readonly string helpText;
         private readonly bool hidden;
+        private readonly bool isDefault;
 
-        public Verb(string name, string helpText, bool hidden = false)
+        public Verb(string name, string helpText, bool hidden = false, bool isDefault = false)
         {
-            this.name = name ?? throw new ArgumentNullException(nameof(name));
+            if (!isDefault && string.IsNullOrWhiteSpace(name))
```
Comment (moh-hassan) [inline src/CommandLine/Core/Verb.cs]: //name should not be null or empty independent of isDefault
```suggestion
            if ( string.IsNullOrWhiteSpace(name))
```

File: src/CommandLine/Core/Verb.cs [hunk 3]
```diff
@@ -12,12 +12,17 @@ sealed class Verb
...
         private readonly bool hidden;
+        private readonly bool isDefault;
 
-        public Verb(string name, string helpText, bool hidden = false)
+        public Verb(string name, string helpText, bool hidden = false, bool isDefault = false)
         {
-            this.name = name ?? throw new ArgumentNullException(nameof(name));
+            if (!isDefault && string.IsNullOrWhiteSpace(name))
+                throw new ArgumentNullException(nameof(name));
+            this.name = name ?? string.Empty;
```
Comment (moh-hassan) [inline src/CommandLine/Core/Verb.cs]:
```suggestion
            this.name = name;
```

File: src/CommandLine/VerbAttribute.cs [hunk 4]
```diff
@@ -12,19 +12,22 @@ namespace CommandLine
...
 
-            this.name = name;
+            this.name = name ?? string.Empty;
```
Comment (moh-hassan) [inline src/CommandLine/VerbAttribute.cs]:
```suggestion
            this.name = name ;
```

**Comment by moh-hassan** [comment]
It's nice if you git rebase to origin\develop  to sync the last changes in develop branch.

**Comment by Artentus** [comment]
Sorry it took me so long, have been busy this week, but everything is merged now.

I see you don't want to allow default verbs with no name. My reasoning behind this was to just have it behave like no verb at all. Functionally it doesn't change anything tho.

**Review by moh-hassan — APPROVED** [review]

**Comment by moh-hassan** [comment]
Thanks @Artentus  for Update.
PR is approved and ready to merge

**Comment by moh-hassan** [comment]
Merged manually to resolve merge confliction at [commit 4bed3432c](https://github.com/commandlineparser/commandline/commit/4bed3432c582b0591b98b6f8b15d5572c8df7055)

**Pull request closed without merging**